app/grandchallenge/challenges/views.py

```diff
@@ -252,9 +252,7 @@ class ChallengeRequestBudgetUpdate(
 
     def form_valid(self, form):
         super().form_valid(form)
-        response = HttpResponse()
-        response["HX-Refresh"] = "true"
-        return response
+        return HttpResponse()
```

Comment:
I didn't dive too deep into the reason why a new HttpResponse was generated in several locations in the first place. It seems to prevent returning a redirect response from the `FormMixin`. The goal was to replicate the behavior it had with a tiny clean-up with respect to the header setting. I hadn't thought about a 200 reply of a form representing a validation error, does it?

I opted for dispatch as it also allows use for the actual place I wrote it for, namely the `rest_framework.DestroyModelMixin`. But that is based on a very quick glance at the subclasses involved. Perhaps you have a good alternative?

Reply:
Yes, on the web a 200 response to a POST tells you that something went wrong. POST requests should always redirect, unless it's an API response. Here you're trying to do something that handles both, Django Class Base Views and Django Rest Framework are separate things, and it is going to be really difficult to write something that handles both.

Honestly, if I were you I would fix the bug using the existing tools and not try to refactor all of this, it may be best to start over.

Reply:
I agree. Thanks for the honest reply. This refactor has gone off the rails. I'll consider this a warm-up for some more Django-tinted work.

Reply:
Yes, I thought you were only tackling Django Template Views here, mixing in HTMX (which is for delivering HTML over the wire) with Django Rest Framework (which is for delivering JSON (principally) over the wire) also makes it way more complicated than it needs to be. If you wanted to make this view HTMX then you would first need to take it out of DRF and put it in a Django Template View, then make it HTMX. But yes, too much work for no user-benefit.